### ISSUE TITLE
PNG output rather than SVG in emacs-jupyter

### ISSUE THREAD

**Issue by GregorySchwartz** [gibiansky/IHaskell#877]
I'm not sure if this is an IHaskell issue or an ob-ipython issue, but with `:opt svg` the output is still a png file.

```
#+HEADER: :results raw drawer
#+BEGIN_SRC jupyter-haskell
:extension NoMonomorphismRestriction FlexibleContexts TypeFamilies
:opt svg

import Diagrams.Prelude
import qualified Diagrams.Prelude as D
import Diagrams.Backend.SVG

-- By Brent Yorgey
-- Draw a Sierpinski triangle!
sierpinski 1 = eqTriangle 1
sierpinski n =     s
                  ===
               (s ||| s) D.# centerX
  where s = sierpinski (n-1)

-- The `diagram` function is used to display them in the notebook.
diagram $ sierpinski 4
            D.# centerXY
            D.# fc black
          `atop` square 10
                   D.# fc white
#+END_SRC
```

**Comment by vaibhavsagar** [comment]
Are you still experiencing this? It looks like the [relevant code](https://github.com/gibiansky/IHaskell/blob/ff046bf2c4c1c1c4efb54a692c4f20c9d5c113f0/ihaskell-display/ihaskell-diagrams/IHaskell/Display/Diagrams.hs#L18-L21) outputs both PNG and SVG.

**Comment by GregorySchwartz** [comment]
I am. I do want to reiterate that it's through https://github.com/dzop/emacs-jupyter now. It works on the browser notebook, but not in the org-mode run. However, I don't see why it would be any different. Using the `jupyter-python` code blocks, I can get both `png` by default and `svg` with `%config InlineBackend.figure_format = 'svg'`, so I don't see how it would be different. Are there inconsistencies with reporting `svg` and `png` between these kernels?

**Comment by GregorySchwartz** [comment]
Okay, see https://github.com/dzop/emacs-jupyter/issues/16#issuecomment-465728972 for the reason and a possible solution. It appears to be because the ihaskell kernel embeds the image in html. A possible solution is to provide it as a `image/svg+xml` similar to what IJulia and IPython do (which he says is according to the Jupyter spec).

**Comment by vaibhavsagar** [comment]
I'm not sure what you mean by "embeds the image in html". Would you be able to point me to the relevant part of the code where this is happening?

**Comment by GregorySchwartz** [comment]
I tried searching and I'm having difficulty finding where this information gets displayed in the code.

**Comment by GregorySchwartz** [comment]
The front end `emacs-jupyter` developer said: "It looks like the haskell kernel is giving an svg image embedded in html which is why you are seeing only png. I think it would be better to ask the people who make the Haskell kernel to provide an image/svg+xml representation of their images before doing anything on this side."

**Comment by GregorySchwartz** [comment]
Could this area be the offender? https://github.com/gibiansky/IHaskell/blob/027f74aa1f6e007feec1650bbaf0d797815be157/src/IHaskell/Publish.hs#L72-L78

**Comment by GregorySchwartz** [comment]
Does base64 need to be used here rather than plain text svg? What is the purpose of using the `html` function, is that was jupyter specifies?

**Comment by GregorySchwartz** [comment]
I removed `convertSvgToHtml` and everything works on the browser and in emacs. Perhaps it was legacy, as `DisplayData MimeSvg` is fine, no need for html. Either way, I'll submit a pull request for further testing in other cases.

**Comment by vaibhavsagar** [comment]
Merged #1009, thanks again!
